cmd/ndm_daemonset/probe/queuesysfsprobe.go

```diff
@@ -0,0 +1,137 @@
+/*
+Copyright 2020 OpenEBS Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+
+Find queue sysfs doc here.
+https://www.kernel.org/doc/Documentation/block/queue-sysfs.txt
+
+*/
+
+package probe
+
+import (
+	"io/ioutil"
+	"strconv"
+	"strings"
+
+	"github.com/openebs/node-disk-manager/cmd/ndm_daemonset/controller"
+	"github.com/openebs/node-disk-manager/pkg/util"
+	"k8s.io/klog"
+)
+
+const (
+	queuesysfsProbePriority = 5
+	queuesysfsConfigKey     = "queuesysfs-probe"
+)
+
+var (
+	queuesysfsProbeName  = "queuesysfs probe"
+	queuesysfsProbeState = defaultEnabled
+)
+
+var queuesysfsProbeRegister = func() {
+	ctrl := <-controller.ControllerBroadcastChannel
+	if ctrl == nil {
+		klog.Error("unable to configure", queuesysfsProbeName)
+		return
+	}
+	if ctrl.NDMConfig != nil {
+		for _, probeConfig := range ctrl.NDMConfig.ProbeConfigs {
+			if probeConfig.Key == queuesysfsConfigKey {
+				queuesysfsProbeName = probeConfig.Name
+				queuesysfsProbeState = util.CheckTruthy(probeConfig.State)
+				break
+			}
+		}
+	}
+
+	newRegistryProbe := &registerProbe{
+		priority:   queuesysfsProbePriority,
+		name:       queuesysfsProbeName,
+		state:      queuesysfsProbeState,
+		pi:         newQueuesysfsProbe(),
+		controller: ctrl,
+	}
+	newRegistryProbe.register()
+}
+
+// queuesysfsProbe fills the logical sector size,
+// physical sector size, drive type(ssd or hdd) of the disk
+type queuesysfsProbe struct {
+}
+
+func newQueuesysfsProbe() *queuesysfsProbe {
+	return &queuesysfsProbe{}
+}
+
+// It is part of probe interface. Hence, empty implementation.
+func (cp *queuesysfsProbe) Start() {}
+
+// FillDiskDetails updates the logical sector size,
+// physical sector size, drive type(ssd or hdd) of the disk
+// if those are not populated.
+func (cp *queuesysfsProbe) FillDiskDetails(d *controller.DiskInfo) {
+	sysPath := d.ProbeIdentifiers.UdevIdentifier
+
+	if d.LogicalSectorSize == 0 {
+		logicalSectorSize, err := readQueuesysfsFileAsInt64(sysPath + "/queue/logical_block_size")
+		if err != nil {
+			klog.Error("unable to read logical block size", err)
```

Comment:
let this be a warning, as these details are not mandatory for functioning.

Reply:
done :+1: